client/client_test.go

```diff
@@ -781,6 +782,178 @@ func TestProgressiveCall(t *testing.T) {
+	ctx := context.Background()
+
+	callSends := 0
+	sendProgDataCb := func(ctx context.Context) (options wamp.Dict, args wamp.List, kwargs wamp.Dict, err error) {
+		options = wamp.Dict{}
+
+		if callSends == (len(callArgs) - 1) {
+			options[wamp.OptProgress] = false
+		} else {
+			options[wamp.OptProgress] = true
+		}
+
+		args = wamp.List{callArgs[callSends]}
+		callSends++
+
+		time.Sleep(100 * time.Millisecond)
+		return options, args, nil, nil
+	}
+
+	result, err := caller.CallProgressive(ctx, procName, sendProgDataCb, nil)
+	if err != nil {
+		t.Fatal("Failed to call procedure:", err)
+	}
+	sum, ok := wamp.AsInt64(result.Arguments[0])
+	if !ok {
+		t.Fatal("Could not convert result to int64:", result.Arguments[0])
+	}
+	if sum != 55 {
+		t.Fatal("Wrong result:", sum)
+	}
+	if callSends != 10 {
+		t.Fatal("Expected callSends == 10")
+	}
+
+	// Test unregister.
+	if err = callee.Unregister(procName); err != nil {
+		t.Fatal("Failed to unregister procedure:", err)
+	}
+
+	caller.Close()
+	callee.Close()
+	r.Close()
+}
+
+func TestProgressiveCallsAndResults(t *testing.T) {
+	// Connect two clients to the same server
+	callee, caller, r, err := connectedTestClients()
+	if err != nil {
+		t.Fatal("failed to connect test clients:", err)
+	}
+
+	// Handler sends progressive results.
+	invocationHandler := func(ctx context.Context, inv *wamp.Invocation) InvokeResult {
+
+		if isInProgress, _ := inv.Details[wamp.OptProgress].(bool); !isInProgress {
+			return InvokeResult{Args: inv.Arguments}
+		} else {
+			senderr := callee.SendProgress(ctx, inv.Arguments, nil)
+			if senderr != nil {
+				callee.log.Println("Error sending progress:", senderr)
+				return InvokeResult{Err: "test.failed"}
+			}
+		}
+
+		return InvokeResult{Err: wamp.InternalProgressiveOmitResult}
+	}
+
+	procName := "nexus.test.progproc"
+
+	// Register procedure
+	if err = callee.Register(procName, invocationHandler, nil); err != nil {
+		t.Fatal("Failed to register procedure:", err)
+	}
+
+	// Test calling the procedure.
+	callArgs := wamp.List{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
+	ctx := context.Background()
+
+	var progressiveResults []int
+	var mu sync.Mutex
+
+	progRescb := func(result *wamp.Result) {
+		mu.Lock()
+		progressiveResults = append(progressiveResults, result.Arguments[0].(int))
+		mu.Unlock()
+	}
+
+	callSends := 0
+	sendProgDataCb := func(ctx context.Context) (options wamp.Dict, args wamp.List, kwargs wamp.Dict, err error) {
+		options = wamp.Dict{}
+
+		if callSends == (len(callArgs) - 1) {
+			options[wamp.OptProgress] = false
+		} else {
+			options[wamp.OptProgress] = true
+		}
+
+		args = wamp.List{callArgs[callSends]}
+		callSends++
+
+		time.Sleep(100 * time.Millisecond)
+		return options, args, nil, nil
+	}
+
+	result, err := caller.CallProgressive(ctx, procName, sendProgDataCb, progRescb)
+	if err != nil {
+		t.Fatal("Failed to call procedure:", err)
+	}
+	progressiveResults = append(progressiveResults, result.Arguments[0].(int))
```

Comment:
same thing about verifying the order of callArgs and progressiveResults to match